noobaa_sa/s3_client.py

```diff
@@ -241,10 +334,82 @@ def delete_object(self, bucket_name, object_key):
         """
         Delete an object from an S3 bucket using boto3
 
+        Args:
+            bucket_name (str): The name of the bucket
+            object_key (str): The key of the object
+
         """
         log.info(f"Deleting object {object_key} from bucket {bucket_name} via boto3")
         self._boto3_client.delete_object(Bucket=bucket_name, Key=object_key)
 
+    def delete_objects(self, bucket_name, object_keys, quiet=True):
+        """
+        Delete multiple objects from an S3 bucket using boto3
+
+        Args:
+            bucket_name (str): The name of the bucket
+            object_keys (list): A list of the keys of the objects to delete
+            quiet (bool): Should the response not contain the result of each delete operation
+
+        Returns:
+            dict: A dictionary containing the response from the delete operation
+
+        """
+        log.info(
+            f"Deleting {len(object_keys)} objects from bucket {bucket_name} via boto3"
+        )
+        delete_objects_dict = {
+            "Objects": [{"Key": key} for key in object_keys],
+            "Quiet": quiet,
+        }
+        response = self._boto3_client.delete_objects(
+            Bucket=bucket_name, Delete=delete_objects_dict
+        )
+        return response
+
+    def copy_object(self, src_bucket, src_key, dest_bucket, dest_key):
+        """
+        Copy an object using boto3
+
+        Args:
+            src_bucket (str): The name of the source bucket
+            src_key (str): The key of the source object
+            dest_bucket (str): The name of the destination bucket
+            dest_key (str): The key of the destination object
+
+        Returns:
+            dict: A dictionary containing the response from the copy operation
+
+        Raises:
+            NoSuchBucket: If the source or destination bucket does not exist
+            NoSuchKey: If the source object does not exist
+            ClientError: If an unexpected error occurs
+
+        """
+        log.info(
+            f"Copying object {src_key} from {src_bucket} to {dest_bucket}/{dest_key}"
+        )
+        try:
+            response = self._boto3_client.copy_object(
+                Bucket=dest_bucket,
+                CopySource={"Bucket": src_bucket, "Key": src_key},
+                Key=dest_key,
+            )
+            return response
+        except ClientError as e:
+            if e.response["Error"]["Code"] == "NoSuchBucket":
+                log.warn(
+                    f"Source or destination bucket does not exist: {e.response['Error']['Message']}"
+                )
+                raise NoSuchBucket(e)
+            elif e.response["Error"]["Code"] == "NoSuchKey":
+                log.warn(
+                    f"Source object {src_key} does not exist in bucket {src_bucket}"
+                )
+                raise NoSuchKey(e)
+            else:
+                raise e
```

Comment:
Same for this as well

Reply:
Done in https://github.com/red-hat-storage/noobaa-sa-ci/pull/16/commits/cde85fc0c8aae17993097f4a72b721a00cc42873